src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js

```diff
@@ -266,9 +266,10 @@ angular.module('esn.inbox.libs')
       return withJmapClient(function(jmapClient) {
         return jmapClient.mailbox_changes({
           accountId: null,
-          sinceState: inboxMailboxesCache.state
+          sinceState: inboxMailboxesCache.state,
+          maxChanges: MAILBOX_MAXIMUM_CHANGES_RESPONSE
         }).then(function(changes) {
-          if (changes.newState !== inboxMailboxesCache.state) {
+          if (changes.hasMoreChanges) {
```

Comment:
This sems wrong. Even if `hasMoreChanges` is false, if the state has changed we should fetch mailboxes again.

Reply:
so we should only compare the state diff

Reply:
At least.
You can even take into account this issue: https://github.com/iNPUTmice/jmap/issues/47
If `hasMoreChanges` is true, anyway we can fetch all mailboxes (even if the state is unchanged)
And to go further, we can treat the error `cannotCalculateChanges` to fetch all mailboxes too.